.github/workflows/contrib-openai.yml

```diff
@@ -334,3 +334,42 @@ jobs:
           with:
             file: ./coverage.xml
             flags: unittests
+
+  AgentOptimizer:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        python-version: [ "3.11" ]
+    runs-on: ${{ matrix.os }}
+    environment: openai1
+    steps:
+      # checkout to pr branch
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install packages and dependencies
+        run: |
+          docker --version
+          python -m pip install --upgrade pip wheel
+          pip install -e .
+          python -c "import autogen"
+          pip install coverage pytest-asyncio
```

Comment:
do we need asyncio?

Reply:
`AgentOptimizer` doesn't have any async code, safe to just install `pytest`